src/MySqlConnector/MySqlConnection.cs

```diff
@@ -917,6 +920,7 @@ internal void Cancel(ICancellableCommand command, int commandId, bool isCancel)
 
 			using var connection = CloneWith(csb.ConnectionString);
 			connection.m_connectionSettings = connectionSettings;
+			connection.ConnectionOpenedCallback = null; // clear the callback because the user doesn't need to execute any setup logic on this connection
```

Comment:
How do we know that?

Reply:
It's a non-pooled connection that will execute `KILL QUERY` and be closed immediately. This is more of an implementation detail than a user-visible `MySqlConnection`. I can reword the comment.